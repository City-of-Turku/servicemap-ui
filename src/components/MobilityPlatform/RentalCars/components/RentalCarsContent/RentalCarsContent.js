import React from 'react';
import PropTypes from 'prop-types';
import { Link, Typography } from '@material-ui/core';

const RentalCarsContent = ({ classes, intl, car }) => {
  const titleText = (messageId, props = {}) => (
    <div className={classes.title}>
      <Typography variant="subtitle1" {...props}>
        {intl.formatMessage({
          id: messageId,
        })}
      </Typography>
    </div>
  );

  const contentText = (messageId, text) => (
    <div className={classes.text}>
      <Typography variant="body2">
        <strong>
          {intl.formatMessage({
            id: messageId,
          })}
          :
        </strong>
        {' '}
        {text}
      </Typography>
    </div>
  );

  const renderCarInfo = (messageId, manufacturer, model) => (
    <div className={classes.text}>
      <Typography variant="body2">
        <strong>
          {intl.formatMessage({
            id: messageId,
          })}
          :
        </strong>
        {' '}
        {manufacturer}
        {' '}
        {model}
      </Typography>
    </div>
  );

  const serviceProvider = '24Rent';

  return (
    <div className={classes.container}>
      {titleText('mobilityPlatform.content.rentalCars.title')}
      {contentText('mobilityPlatform.content.rentalCars.provider', serviceProvider)}
      <div className={classes.linkContainer}>
        <Link target="_blank" href={`https://www.24rent.fi/#/?city=${car.homeLocationData.fullAddress}`}>
          <Typography className={classes.link} variant="body2">
            {intl.formatMessage({
              id: 'mobilityPlatform.content.rentalCars.link',
            })}

          </Typography>
        </Link>
      </div>
      {renderCarInfo(
        'mobilityPlatform.content.rentalCars.carInfo',
        car.vehicleModelData.manufacturer,
        car.vehicleModelData.name,
      )}
      <div className={classes.text}>
        <Typography>
          {car.availabilityData.available
            ? intl.formatMessage({ id: 'mobilityPlatform.content.rentalCars.available' })
            : intl.formatMessage({ id: 'mobilityPlatform.content.rentalCars.reserved' })}
        </Typography>
      </div>
      {contentText('mobilityPlatform.content.rentalCars.address', car.homeLocationData.fullAddress)}
      <div>
        <img src={`https://vehicles-cdn.24rent.fi/${car.id}_medium.jpeg`} alt="shared use car" />
      </div>
    </div>
  );
};

RentalCarsContent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  car: PropTypes.objectOf(PropTypes.any),
};

RentalCarsContent.defaultProps = {
  car: null,
};

export default RentalCarsContent;
