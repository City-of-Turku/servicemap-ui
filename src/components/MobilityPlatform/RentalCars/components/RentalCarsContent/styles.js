export default theme => ({
  container: {
    margin: theme.spacing(1),
  },
  title: {
    width: '94%',
    borderBottom: '1px solid #000000',
    marginBottom: theme.spacing(1),
  },
  text: {
    paddingBottom: theme.spacing(1),
  },
  linkContainer: {
    paddingBottom: theme.spacing(1),
    width: '55%',
  },
  link: {
    color: '#0000EE',
  },
});
