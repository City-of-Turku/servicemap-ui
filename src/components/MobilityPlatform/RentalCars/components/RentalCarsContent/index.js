import { withStyles } from '@mui/styles';
import { injectIntl } from 'react-intl';
import RentalCarsContent from './RentalCarsContent';
import styles from './styles';

export default withStyles(styles)(injectIntl(RentalCarsContent));
