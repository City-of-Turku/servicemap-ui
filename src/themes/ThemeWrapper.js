import { ThemeProvider } from '@mui/material/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { useSelector } from 'react-redux';
import themes from '.';
import config from '../../config';
import { selectThemeMode } from '../redux/selectors/user';

// If external theme (by Turku) is true, then can be used to select which theme to get
const externalTheme = config.themePKG;
const isExternalTheme = !externalTheme || externalTheme === 'undefined' ? null : externalTheme;

/**
 * Returns default theme based on env value
 * @returns theme
 */
const getTheme = () => {
  if (isExternalTheme) {
    return themes.SMThemeTku;
  }
  return themes.SMTheme;
};

// Component to handle theme changes
const ThemeHandler = ({ children }) => {
  const themeMode = useSelector(selectThemeMode);
  return (// Get correct theme setting from store
    <ThemeProvider theme={themeMode === 'dark' ? themes.SMThemeDark : getTheme()}>
      {children}
    </ThemeProvider>
  );
};
ThemeHandler.propTypes = {
  children: PropTypes.node,
};
ThemeHandler.defaultProps = {
  children: null,
};

export default ThemeHandler;
