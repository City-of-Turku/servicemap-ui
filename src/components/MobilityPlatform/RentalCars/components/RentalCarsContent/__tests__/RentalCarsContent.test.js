/* eslint-disable max-len */
// Link.react.test.js
import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import themes from '../../../../../../themes';
import RentalCarsContent from '../index';
import finnishTranslations from '../../../../../../i18n/fi';

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnishTranslations,
};

const mockProps = {
  car: {
    homeLocationData: {
      fullAddress: 'Testiosoite',
    },
    vehicleModelData: {
      name: 'Testiauto',
      manufacturer: 'Testimerkki',
    },
    availabilityData: {
      available: true,
    },
    link: 'https://www.24rent.fi',
  },
};

// eslint-disable-next-line react/prop-types
const Providers = ({ children }) => (
  <IntlProvider {...intlMock}>
    <MuiThemeProvider theme={themes.SMTheme}>{children}</MuiThemeProvider>
  </IntlProvider>
);

const renderWithProviders = component => render(component, { wrapper: Providers });

describe('<RentalCarsContent />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<RentalCarsContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<RentalCarsContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0]).toBeInTheDocument();
    expect(p[1].textContent).toEqual(mockProps.car.link);
    expect(p[2].textContent).toContain(
      `${finnishTranslations['mobilityPlatform.content.rentalCars.carInfo']}: ${mockProps.car.vehicleModelData.manufacturer} ${mockProps.car.vehicleModelData.name}`,
    );
    expect(p[3].textContent).toContain(finnishTranslations['mobilityPlatform.content.rentalCars.available']);
    expect(p[4].textContent).toContain(
      `${finnishTranslations['mobilityPlatform.content.rentalCars.address']}: ${mockProps.car.homeLocationData.fullAddress}`,
    );
  });

  it('does show link correctly', () => {
    const { container } = renderWithProviders(<RentalCarsContent {...mockProps} />);

    const link = container.querySelector('a');
    expect(link).toBeInTheDocument();
  });

  it('does show image correctly', () => {
    const { container } = renderWithProviders(<RentalCarsContent {...mockProps} />);

    const image = container.querySelector('img');
    expect(image).toBeInTheDocument();
  });
});
