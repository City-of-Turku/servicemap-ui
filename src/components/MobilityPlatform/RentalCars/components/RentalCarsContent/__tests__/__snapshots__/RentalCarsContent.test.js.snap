// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RentalCarsContent /> should work 1`] = `
<div>
  <div
    class="injectIntl(RentalCarsContent)-container-1"
  >
    <div
      class="injectIntl(RentalCarsContent)-title-2"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-dlbwt1-MuiTypography-root"
      >
        Yhteiskäyttöauto
      </h6>
    </div>
    <div
      class="injectIntl(RentalCarsContent)-text-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Palveluntarjoaja: 24Rent
      </p>
    </div>
    <div
      class="injectIntl(RentalCarsContent)-linkContainer-4"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-fixjtq-MuiTypography-root-MuiLink-root"
        href="https://www.24rent.fi/#/?city=Testiosoite"
        target="_blank"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 injectIntl(RentalCarsContent)-link-5 css-1ak20dk-MuiTypography-root"
        >
          https://www.24rent.fi
        </p>
      </a>
    </div>
    <div
      class="injectIntl(RentalCarsContent)-text-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Auton tiedot
        :
         
        Testimerkki
         
        Testiauto
      </p>
    </div>
    <div
      class="injectIntl(RentalCarsContent)-text-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Vapaa auto
      </p>
    </div>
    <div
      class="injectIntl(RentalCarsContent)-text-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-1ak20dk-MuiTypography-root"
      >
        Sijainti: Testiosoite
      </p>
    </div>
    <div>
      <img
        alt="shared use car"
        src="https://vehicles-cdn.24rent.fi/undefined_medium.jpeg"
      />
    </div>
  </div>
</div>
`;
